base/card-api.gts

```diff
@@ -256,13 +257,15 @@ class ContainsMany<FieldT extends CardConstructor> implements Field<FieldT> {
     );
   }
 
-  emptyValue(instance: Card) { return new WatchedArray(() => recompute(instance)) }
+  emptyValue(instance: Card) {
+    return new WatchedArray( () => instance[myRecomputes].push(recompute(instance)));
```

Comment:
we can push these promises directly into the logger instead of having the card instance keep track